accord-core/src/main/java/accord/coordinate/AbstractCoordinatePreAccept.java

```diff
@@ -219,7 +220,7 @@ final void onNewEpoch(Topologies prevTopologies, long latestEpoch)
                 onNewEpochTopologyMismatch(mismatch);
                 return;
             }
-            topologies = node.topology().withUnsyncedEpochs(route, txnId.epoch(), latestEpoch);
+            topologies = node.topology().withUnsyncedEpochs(route, txnId == null ? executeAtEpoch() : txnId.epoch(), latestEpoch);
```

Comment:
Perhaps introduce a helper method `earliestEpoch` or something (that could in theory be overridden) so this semantic is clearer?